terminator/src/platforms/windows.rs

```diff
@@ -2234,6 +2237,115 @@ impl UIElementImpl for WindowsUIElement {
+                Ok(monitor) => match monitor.scale_factor() {
+                    Ok(factor) => factor as f64,
+                    Err(e) => {
+                        error!("Failed to get scale factor from cursor position: {}", e);
+                        1.0 // Fallback to default scale factor
+                    }
+                },
+                Err(e) => {
+                    error!("Failed to get monitor from cursor position: {}", e);
+                    1.0 // Fallback to default scale factor
+                }
+            }
+        }
+
+        // Helper function to get scale factor from focused window
+        fn get_scale_factor_from_focused_window() -> Option<f64> {
+            match xcap::Window::all() {
+                Ok(windows) => {
+                    windows.iter()
+                        .find(|w| w.is_focused().unwrap_or(false))
+                        .and_then(|focused_window| {
+                            focused_window.current_monitor().ok()
+                        })
+                        .and_then(|monitor| {
+                            monitor.scale_factor().ok().map(|factor| factor as f64)
+                        })
+                },
+                Err(e) => {
+                    error!("Failed to get windows: {}", e);
+                    None
+                }
+            }
+        }
+
+        // Try to get scale factor from focused window first, fall back to cursor position
+        let scale_factor = get_scale_factor_from_focused_window()
+            .unwrap_or_else(get_scale_factor_from_cursor);
+
+        // Constants for border appearance
+        const BORDER_SIZE: i32 = 4;
+        const DEFAULT_RED_COLOR: u32 = 0x0000FF; // Pure red in BGR format
+
+        // Use provided color or default to red
+        let highlight_color = color.unwrap_or(DEFAULT_RED_COLOR);
+
+        // Scale the coordinates and dimensions
+        let x = (rect.get_left() as f64 * scale_factor) as i32;
+        let y = (rect.get_top() as f64 * scale_factor) as i32;
+        let width = (rect.get_width() as f64 * scale_factor) as i32;
+        let height = (rect.get_height() as f64 * scale_factor) as i32;
+
+        // Get the screen DC
+        let hdc = unsafe { GetDC(None) };
+        if hdc.0.is_null() {
+            return Err(AutomationError::PlatformError(
+                "Failed to get device context".to_string(),
+            ));
+        }
+
+        unsafe {
+            // Create a pen for drawing with the specified color
+            let hpen = CreatePen(PS_SOLID, BORDER_SIZE, COLORREF(highlight_color));
+            if hpen.0.is_null() {
+                ReleaseDC(None, hdc);
+                return Err(AutomationError::PlatformError(
+                    "Failed to create pen".to_string(),
+                ));
+            }
+
+            // Save current objects
+            let old_pen = SelectObject(hdc, HGDIOBJ(hpen.0));
+            let null_brush = GetStockObject(NULL_BRUSH);
```

Comment:
i dont get it


i suggest you provide a param like duration and this would spawn a thread so it can turn on / off the highlight without blocking the main thread